### DISCUSSION TITLE
Autocenter via command line

### DISCUSSION THREAD

**Discussion by bondbenz** [sumatrapdfreader/sumatrapdf#4385] [General]
Hello :)

Is there a way to enable auto center or disable it via the command line? (a slient print) that I am willing to use inside my code, thank you.

**Comment by GitHubRulesOK** [comment]
Hmm I am not sure KJK (The developer) will agree but Printing is not "SumatraPDF Reader"'s forte. It now relies heavily on Windows "Print As Image" system with all its problems. However AFAIK Auto Centering is not an exposed method it is down to drivers to offer their capabilities and I think that is individualistic.

I trawleed Microsoft KnowledgBase looking for Printer Auto-center/autocenter and there are AFAIK 2 related data structures 
- Display Monitor  (DM related)
- Scan is centered on bed

There seem to be no driver controls for print positioning (unless possibly in a manufacturers repertoire).

 By default <strike>"Fit"</strike>? &  "Shrink" should center page but others use image datum of top left unless the driver has multiple custom profiles. (you can set your own offsets via marginal offsets in windows "forms")
![image](https://github.com/user-attachments/assets/3c076e31-3569-4847-b6f3-0544f0bf1c7c)


output A5 image as "shrink" on A4 media
![image](https://github.com/user-attachments/assets/4f2a4909-bc7c-4c51-a2dd-4d1121335d1c)
